### PR TITLE
validation-search

### PR DESCRIPTION
**Pull request by VictorYarosh**

### CODE REVIEW

File: src/components/ui/weather-card-control/use-weather-card-control.ts [hunk 1]
```diff
@@ -1,28 +1,43 @@
...
+          setCities([...cities, values.search]);
+          setIsActive(false);
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    addWeather().catch((error) => {
+      console.error(error);
+    });
```
Comment (toserg) [inline src/components/ui/weather-card-control/use-weather-card-control.ts]: redundant

File: src/components/ui/weather-card-control/use-weather-card-control.ts [hunk 2]
```diff
@@ -1,28 +1,43 @@
...
-  const [isLoading, setLoading] = useState(true);
-  const { setCities, cities } = useContext(CardsContext);
+  const [loadingCity, setLoadingCity] = useState(true);
+  const { cities, setCities } = useContext(CardsContext);
 
   const handleAddSearch = () => {
     setIsActive(true);
   };
 
   const handleOnSubmit = ({ values }: any) => {
```
Comment (toserg) [inline src/components/ui/weather-card-control/use-weather-card-control.ts]: add async

File: src/components/ui/weather-card-control/use-weather-card-control.ts [hunk 3]
```diff
@@ -1,28 +1,43 @@
...
-    setCities([...cities, values.search]);
-    setIsActive(false);
+    const addWeather = async () => {
```
Comment (toserg) [inline src/components/ui/weather-card-control/use-weather-card-control.ts]: remove the redundant wrap function, leave only try/catch

File: src/components/ui/weather-card/use-weather-card.ts [hunk 4]
```diff
@@ -27,7 +29,15 @@ const useWeatherCard = ({ city }: WeatherCardProps) => {
...
   }, []);
 
-  return { dateBuilder, data, isLoading };
+  const handleDeleteCard = () => {
+    setCities(
+      cities.filter(function (value, index, array) {
+        return index === 0;
+      })
+    );
+  };
```
Comment (toserg) [inline src/components/ui/weather-card/use-weather-card.ts]: Your handle delete function doesn't work. Your filter function removes all cities from the array except the first city. Please update the logic.

**Review by toserg — CHANGES_REQUESTED** [review]

File: src/components/ui/weather-card-control/use-weather-card-control.ts (hunk 1 above)
Comment (VictorYarosh) [reply src/components/ui/weather-card-control/use-weather-card-control.ts]: fixed

File: src/components/ui/weather-card-control/use-weather-card-control.ts (hunk 3 above)
Comment (VictorYarosh) [reply src/components/ui/weather-card-control/use-weather-card-control.ts]: fixed

File: src/components/ui/weather-card/use-weather-card.ts (hunk 4 above)
Comment (VictorYarosh) [reply src/components/ui/weather-card/use-weather-card.ts]: fixed

File: src/components/ui/weather-card-control/use-weather-card-control.ts (hunk 2 above)
Comment (VictorYarosh) [reply src/components/ui/weather-card-control/use-weather-card-control.ts]: fixed

**Review by toserg — APPROVED** [review]

**Pull request merged**